src/Tests/UnitTest/TestComponentCSharp_Tests.cs

```diff
@@ -2219,6 +2222,15 @@ public void TestStringUnboxing()
             Assert.IsType<string>(str2);
             Assert.Equal(string.Empty, (string)str1);
             Assert.Equal(string.Empty, (string)str2);
+        }
+
+        [Fact]
+        public void TestDelegateUnboxing()
```

Comment:
Should we add any tests for all the new nullable types we added? Or are all of them being tested already?

Reply:
Added tests which actually also ended up catching an existing issue with IReference<char>.